From 55a0c9f373596016729a10fa44b995886fc4d924 Mon Sep 17 00:00:00 2001
From: Karsten Merker <merker@debian.org>
Date: Thu, 5 Mar 2015 19:57:10 +0100
Subject: [PATCH] Backport support for the LeMaker Banana Pro board from
 v2015.04rc1

This backport is based on the upstream commit
http://git.denx.de/?p=u-boot.git;a=patch;h=81ec10015181bcf19a190ae4bea94bb1ac8d1da6

For use with u-boot 2014.10, some adjustments for differences in
the build systems between u-boot v2014.10 and v2015.04rc1 had to
be made.

Commit log from the upstream commit:
====================================
sun7i: Add new Bananapro board / defconfig

Add support for the new Bananapro A20 development board from lemaker.org.
This board features 1G RAM, 2 USB A receptacles, 1 micro USB receptacle for
OTG, 1 micro USB receptacle for power, HDMI, sata, Gbit ethernet, ir receiver,
3.5 mm jack for a/v out, on board microphone, 40 gpio pins and sdio wifi.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Acked-by: Ian Campbell <ijc@hellion.org.uk>
---
 .../Add-support-for-the-LeMaker-Banana-Pro.patch   | 48 ++++++++++++++++++++++
 debian/patches/series                              |  2 +
 debian/targets                                     |  3 ++
 3 files changed, 53 insertions(+)
 create mode 100644 debian/patches/BananaPro/Add-support-for-the-LeMaker-Banana-Pro.patch

diff --git a/debian/patches/BananaPro/Add-support-for-the-LeMaker-Banana-Pro.patch b/debian/patches/BananaPro/Add-support-for-the-LeMaker-Banana-Pro.patch
new file mode 100644
index 0000000..dfbd1fa
--- /dev/null
+++ b/debian/patches/BananaPro/Add-support-for-the-LeMaker-Banana-Pro.patch
@@ -0,0 +1,48 @@
+Index: u-boot-2014.10+dfsg1/board/sunxi/MAINTAINERS
+===================================================================
+--- u-boot-2014.10+dfsg1.orig/board/sunxi/MAINTAINERS
++++ u-boot-2014.10+dfsg1/board/sunxi/MAINTAINERS
+@@ -19,6 +19,7 @@ F:	configs/r7-tv-dongle_defconfig
+ F:	include/configs/sun7i.h
+ F:	configs/A20-OLinuXino_MICRO_defconfig
+ F:	configs/Bananapi_defconfig
++F:	configs/Bananapro_defconfig
+ F:	configs/i12-tvbox_defconfig
+ F:	configs/Linksprite_pcDuino3_defconfig
+ F:	configs/qt840a_defconfig
+Index: u-boot-2014.10+dfsg1/board/sunxi/Makefile
+===================================================================
+--- u-boot-2014.10+dfsg1.orig/board/sunxi/Makefile
++++ u-boot-2014.10+dfsg1/board/sunxi/Makefile
+@@ -22,6 +22,7 @@ obj-$(CONFIG_A20_OLINUXINO_M)	+= dram_su
+ obj-$(CONFIG_AUXTEK_T004)	+= dram_a10s_olinuxino_m.o
+ obj-$(CONFIG_BA10_TV_BOX)	+= dram_sun4i_384_1024_iow8.o
+ obj-$(CONFIG_BANANAPI)		+= dram_bananapi.o
++obj-$(CONFIG_BANANAPRO) 	+= dram_bananapi.o
+ obj-$(CONFIG_CUBIEBOARD)	+= dram_cubieboard.o
+ obj-$(CONFIG_CUBIEBOARD2)	+= dram_cubieboard2.o
+ obj-$(CONFIG_CUBIETRUCK)	+= dram_cubietruck.o
+Index: u-boot-2014.10+dfsg1/board/sunxi/gmac.c
+===================================================================
+--- u-boot-2014.10+dfsg1.orig/board/sunxi/gmac.c
++++ u-boot-2014.10+dfsg1/board/sunxi/gmac.c
+@@ -29,7 +29,7 @@ int sunxi_gmac_initialize(bd_t *bis)
+ 	 * need to set bits 10-12 GTXDC "GMAC Transmit Clock Delay Chain"
+ 	 * of the GMAC clk register to 3.
+ 	 */
+-#ifdef CONFIG_BANANAPI
++#if defined CONFIG_BANANAPI || defined CONFIG_BANANAPRO
+ 	setbits_le32(&ccm->gmac_clk_cfg, 0x3 << 10);
+ #endif
+ 
+Index: u-boot-2014.10+dfsg1/configs/Bananapro_defconfig
+===================================================================
+--- /dev/null
++++ u-boot-2014.10+dfsg1/configs/Bananapro_defconfig
+@@ -0,0 +1,6 @@
++CONFIG_SPL=y
++CONFIG_SYS_EXTRA_OPTIONS="BANANAPRO,AXP209_POWER,SUNXI_GMAC,RGMII,MACPWR=SUNXI_GPH(23),AHCI,USB_EHCI,SUNXI_USB_VBUS0_GPIO=SUNXI_GPH(0),SUNXI_USB_VBUS1_GPIO=SUNXI_GPH(1)"
++CONFIG_FDTFILE="sun7i-a20-bananapro.dtb"
+++S:CONFIG_ARM=y
+++S:CONFIG_TARGET_SUN7I=y
++
diff --git a/debian/patches/series b/debian/patches/series
index 5c628fc..7b35266 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -42,3 +42,5 @@ arndale/exynos5-Use-config_distro_bootcmd.h.patch
 sh4-fix-linker-name-prefix.patch
 
 A20-OLinuXino-LIME2/sun7i-Add-support-for-Olimex-A20-OLinuXino-LIME2.patch
+
+BananaPro/Add-support-for-the-LeMaker-Banana-Pro.patch
diff --git a/debian/targets b/debian/targets
index d15c470..514a743 100644
--- a/debian/targets
+++ b/debian/targets
@@ -30,6 +30,9 @@ armhf	sunxi		A20-OLinuXino-Lime2	u-boot-sunxi-with-spl.bin
 # Vagrant Cascadian <vagrant@debian.org>
 armhf	sunxi		Bananapi	u-boot-sunxi-with-spl.bin
 
+# Karsten Merker <merker@debian.org>
+armhf	sunxi		Bananapro	u-boot-sunxi-with-spl.bin
+
 # Vagrant Cascadian <vagrant@debian.org>
 armhf	sunxi		Cubieboard	u-boot-sunxi-with-spl.bin
 
-- 
2.1.4

